lxd/device/gpu_sriov.go

```diff
@@ -83,25 +83,49 @@ func (d *gpuSRIOV) Start() (*deviceConfig.RunConfig, error) {
 	saveData := make(map[string]string)
 
 	// Get SRIOV parent, i.e. the actual GPU.
-	parentPCIAddress, err := d.getParentPCIAddress()
+	parentPCIAddresses, err := d.getParentPCIAddresses()
 	if err != nil {
 		return nil, err
 	}
 
-	// Get PCI information about the GPU device.
-	devicePath := filepath.Join("/sys/bus/pci/devices", parentPCIAddress)
+	var parentPCIAddress string
+	var pciParentDev pcidev.Device
+	vfID := -1
 
-	pciParentDev, err := pcidev.ParseUeventFile(filepath.Join(devicePath, "uevent"))
-	if err != nil {
-		return nil, errors.Wrapf(err, "Failed to get PCI device info for GPU %q", parentPCIAddress)
+	// Since there might be multiple GPUs, we iterate through them and get the first free
+	// virtual function.
+	for _, parentPCIAddress = range parentPCIAddresses {
+		// Get PCI information about the GPU device.
+		devicePath := filepath.Join("/sys/bus/pci/devices", parentPCIAddress)
+
+		pciParentDev, err = pcidev.ParseUeventFile(filepath.Join(devicePath, "uevent"))
+		if err != nil {
+			err = errors.Wrapf(err, "Failed to get PCI device info for GPU %q", parentPCIAddress)
+			continue
+		}
+
+		vfID, err = d.findFreeVirtualFunction(pciParentDev)
+		if err != nil {
+			err = errors.Wrap(err, "Failed to find free virtual function")
+			continue
+		}
+
+		if vfID == -1 {
```

Comment:
@monstermunchkin just a nit, but how about changing this to:

```
if vfID > -1 {
   break
}
```

Then lose the break at the end of the loop. It feels more natural (subjective I know) to let the loop continue itself as normal without an explicit continue when still searching, and breaking when the VF is found.